spanner/value.go

```diff
@@ -2995,6 +3007,38 @@ func decodeStructArray(ty *sppb.StructType, pb *proto3.ListValue, ptr interface{
 	return nil
 }
 
+func getAllFieldNames(v reflect.Value) []string {
+	var names []string
+	s := v
```

Comment:
Why do we need this?

Reply:
We need this for cases when GO struct has more than one common spanner tags, In that case we want to return duplicate error. There was no way to get all exported struct field names so created one here. please let me know if you think there is another good way to do this.

Reply:
Sorry, I should have been more specific. I meant literally why do we need this specific line:

`s := v`

Why can't the method just use `v`?